cli/tests/test_git_push.rs

```diff
@@ -38,7 +38,8 @@ fn set_up() -> (TestEnvironment, PathBuf) {
             "git",
             "clone",
             "--config-toml=git.auto-local-branch=true",
-            origin_git_repo_path.to_str().unwrap(),
+            // Git and libgit2 > 1.8 do not allow backslashes in remote names
+            &origin_git_repo_path.to_str().unwrap().replace("\\", "/"),
```

Comment:
nit: might be safer to apply `dunce::simplified()` first. I didn't know the canonicalized Windows path can have forward slashes.

Reply:
>  didn't know the canonicalized Windows path can have forward slashes.

It's madness. Officially, they can't. For `libgit2` and Git, who knows. See also https://github.com/rhysd/path-slash/issues/6.

You can see my draft of a more principled solution in #4188, but I do not like where it is going so far.